lucene/core/src/java/org/apache/lucene/search/FuzzyQuery.java

```diff
@@ -156,9 +160,13 @@ public Automaton toAutomaton() {
 
   @Override
   public void visit(QueryVisitor visitor) {
-    // TODO find some way of consuming Automata
-    if (visitor.acceptField(term.field())) {
-      visitor.visitLeaf(this);
+    if (visitor.acceptField(field)) {
+      if (maxEdits == 0 || prefixLength >= term.text().length()) {
+        visitor.consumeTerms(this, term);
+      } else {
+        visitor.consumeTermsMatching(this, field,
+            new ByteRunAutomaton(toAutomaton(), false, Operations.DEFAULT_MAX_DETERMINIZED_STATES));
```

Comment:
At least we should acknowledge in a comment that this is expensive :-/
At least I could imagine a user solving this by subclassing FuzzyQuery to override toAutomaton and using a cache.

Reply:
Reading back through this, it looks as though FuzzyQuery rebuilds its automata on every rewrite, which is pretty wasteful - I'll open a new issue to follow this up.

Reply:
I opened https://issues.apache.org/jira/browse/LUCENE-9068